bentoml/_internal/configuration/__init__.py

```diff
@@ -45,7 +45,7 @@ def expand_env_var(env_var: str) -> str:
 
 def clean_bentoml_version(bentoml_version: str) -> str:
     post_version = bentoml_version.split("+")[0]
-    match = re.match(r"^(\d+)\.(\d+)\.(\d+)(?:(a|rc)\d)*", post_version)
+    match = re.match(r"^(\d+)\.(\d+)\.(post)?(\d+)(?:(a|rc)\d)*", post_version)
```

Comment:
What is the reason behind this?
You might mean 
```
r"^(\d+)\.(\d+)\.(\d+)(?:(a|rc|post)\d)*
```
BentoML dirty version can't be `1.0.post3`, should always be `1.0.0post3`?

Reply:
When I forked the repo and pulled my fork and installed it using `pip install -e .`, it seems like the version is `0.0.post1856+g1185809.d20220730`. This caused issues when I tried to import bentoml (or do anything with bentoml really).

Reply:
You probably also need to fetch the tags as well.

Reply:
reverted the change, as my bentoml version is as expected after fetching the tags (1.0.2.post2+g1185809f.d20220730)